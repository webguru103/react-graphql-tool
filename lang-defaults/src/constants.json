[
  {
    "id": "upload-label",
    "defaultMessage": "Upload PDF"
  },
  {
    "id": "drag-file-details",
    "defaultMessage": "Drag & drop, or click this area to upload."
  },
  {
    "id": "no-pdf-uploaded",
    "defaultMessage": "Please upload the form base"
  }
]