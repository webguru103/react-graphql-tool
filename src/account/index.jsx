import * as React from 'react';
import LoginLayout from './loginLayout/loginLayout';
import AgentSignup from './agentSignUp/agentSignUp';
import InvitedToSignUpNew from './invitedToSignUp/invitedToSignUpNew';
import InvitedToSignUpExisting from './invitedToSignUp/invitedToSignUpExisting';
import UserActivation from './userActivation';
import ResetPassword from './resetPassword/resetPassword';
import ForgotPassword from './forgotPassword/forgotPassword';
import AcceptInvitation from './acceptInvitation';
import RejectBrokerage from './rejectBrokerage';
import Login from './login/login';

const withLayout = (Component: React.Element) => function LayoutWrapper(props: any) {
  return (
    <LoginLayout>
      <Component {...props} />
    </LoginLayout>
  );
};

export default {
  AgentSignup: withLayout(AgentSignup),
  InvitedToSignUpNew: withLayout(InvitedToSignUpNew),
  InvitedToSignUpExisting: withLayout(InvitedToSignUpExisting),
  UserActivation: withLayout(UserActivation),
  ResetPassword: withLayout(ResetPassword),
  ForgotPassword: withLayout(ForgotPassword),
  AcceptInvitation: withLayout(AcceptInvitation),
  RejectBrokerage: withLayout(RejectBrokerage),
  Login: withLayout(Login),
};

export { default as AccountSettings } from './account-settings';
export { default as Logout } from './logout';
