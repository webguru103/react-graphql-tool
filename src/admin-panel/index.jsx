import * as React from 'react';
import Loadable from 'react-loadable';
import CircularProgress from '@material-ui/core/CircularProgress';
import Landing from './landing/landing';
import AgentDetails from './agent-management/agentDetails';
import AdminDetails from './admin-management/admin-details/adminDetails';
import AdminDashboard from './dashboard/dashboard';
import UnderConstruction from '../shared/underConstruction';

const AdminManagement = Loadable({
  loader: () => import('./admin-management/admins'),
  loading: () => <CircularProgress />,
});
const AgentManagement = Loadable({
  loader: () => import('./agent-management/agents'),
  loading: () => <CircularProgress />,
});
const BrokerageList = Loadable({
  loader: () => import('./brokerage-management/brokerages'),
  loading: () => <CircularProgress />,
});
const BrokerageDetails = Loadable({
  loader: () => import('./brokerage-management/brokerage-details/brokerageDetails'),
  loading: () => <CircularProgress />,
});
const AccountSettings = Loadable({
  loader: () => import('../account/account-settings/accountSettings'),
  loading: () => <CircularProgress />,
});

const withLayout = (Component: React.Element) => function LayoutWrapper(props: any) {
  return (
    <Landing>
      <Component {...props} />
    </Landing>
  );
};

export { default as UsernameBreadCrumb } from './userNameBreadCrumb';
export { default as BrokerageNameBreadCrumb } from './brokerage-management/brokerageNameBreadCrumb';

export default {
  Index: withLayout(AdminDashboard),
  AgentManagement: withLayout(AgentManagement),
  AdminManagement: withLayout(AdminManagement),
  BrokerageDetails: withLayout(BrokerageDetails),
  BrokerageList: withLayout(BrokerageList),
  AccountSettings: withLayout(AccountSettings),
  AgentManagementDetails: withLayout(AgentDetails),
  AdminManagementDetails: withLayout(AdminDetails),
  UnderConstruction: withLayout(UnderConstruction.Admin),
};
