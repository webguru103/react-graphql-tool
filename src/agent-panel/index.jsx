import * as React from 'react';
import Loadable from 'react-loadable';
import CircularProgress from '@material-ui/core/CircularProgress';
import Landing from './landing-agent/landing';
import InitiateSession from './initiate-session/initiateSession';
import SigningSession from './signing-session/signingSession';
import Feedback from './initiate-session/feedback';
import ViewDocuments from './viewDocuments/editorContainer';
import UnderConstruction from './landing-agent/underConstruction';

const Documents = Loadable({
  loader: () => import('./document/document'),
  loading: () => <CircularProgress />,
});
const AccountSettings = Loadable({
  loader: () => import('../account/account-settings/accountSettings'),
  loading: () => <CircularProgress />,
});
const Sessions = (Loadable({
  loader: () => import('./sessions/sessions'),
  loading: () => <CircularProgress />,
}): React.Element);

const withLayout = (Component: React.Element) => function LayoutWrapper(props: any) {
  return (
    <Landing>
      <Component {...props} />
    </Landing>
  );
};

export default {
  Documents: withLayout(Documents),
  Transactions: withLayout(UnderConstruction),
  Clients: withLayout(UnderConstruction),
  AccountSettings: withLayout(AccountSettings),
  Sessions,
  Feedback,
  InitiateSession,
  SigningSession,
  ViewDocuments,
};

export { default as TransactionBreadcrumb } from './document/breadCrumb';
