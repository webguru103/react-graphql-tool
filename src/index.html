<!doctype html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
  <link
    href="https://fonts.googleapis.com/css?family=Work+Sans:400,500,600|Cedarville+Cursive|Dancing+Script|Great+Vibes|Pacifico|Parisienne|Alex+Brush|Allura|Great+Vibes|Marck+Script|Montez|Over+the+Rainbow|Rochester|Zeyada"
    rel="stylesheet" crossorigin="anonymous">
  <!-- jss-insertion-point -->
</head>

<body>
  <div id="mainapp"></div>
</body>

</html>