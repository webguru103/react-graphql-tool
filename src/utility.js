import snakeCase from 'lodash.snakecase';

const compose2 = (f, g) => (...args) => f(g(...args));
export const compose = (...fns: Array<Function>) => fns.reduce(compose2);

/**
 * Use this function to catch async/await errors without
 * using a try catch block in the style of golang error handling.
 * wrap function to be awaited in pro()
 * eg. const [err, data] = await pro(this.props.asycMethod({});
 */
export function pro(promise: Promise<any>) {
  return promise.then(data => [null, data])
    .catch(err => [err]);
}

export function isObject(obj: any) {
  return obj === Object(obj) && Object.prototype.toString.call(obj) !== '[object Array]';
}

// function can be used to check if the value is actually defined
// it excludes 0, for example
export function isDefined(value: any) {
  return value !== null && value !== undefined;
}

/**
 * Pass this function a string or array that refers to a property in a nested object 'nestedObj'
 * eg. 'brokerage.name' or ['brokerage', 'name']
 * and it will resolve the correct field:
 * return nestedObj.brokerage.name;
 */

export function get(obj: Object = {}, path: string | Array<string>, defaultValue: any) {
  const paths = Array.isArray(path) ? path : path.split('.');

  if (paths.length > 1 && (isObject(obj) || Array.isArray(obj)) && obj[paths[0]]) {
    return get(obj[paths[0]], paths.slice(1), defaultValue);
  }

  if (paths.length === 1 && (isObject(obj) || Array.isArray(obj)) && isDefined(obj[paths[0]])) {
    return obj[paths[0]];
  }
  return defaultValue;
}

/**
 + * Can be used as a attributes empty function so that when
 + * an empty function is needed, there is no need to declare another.
 + * For example, may be used in default proptypes to return an empty
 + * function when no function passed in.
 + */
export const noop = (f: any) => f;

export const delay = (ms: number): Promise<void> => new Promise(res => setTimeout(res, ms));

export function randomNegativeInt() {
  return -Math.round(Math.random() * 10000000000);
}

export function randomInt() {
  return Math.round(Math.random() * 10000000000);
}

export function randomId(range: number) {
  return Math.random().toString(36).substring(2, (range + 2));
}

export function isEmailValid(input: string) {
  const re = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
  return re.test(String(input).toLowerCase());
}

export function parseEmailsFromString(value: string): ?Array<string> {

  const splitEmails = value.trim().split(/[\s,;\t\n]+/);

  if (splitEmails.every((splitEmail: string) => isEmailValid(splitEmail))) {
    return Array.from(new Set(splitEmails));
  }
  return null;
}

export function generateFilterValues(filters: Array<{ [string]: string }>) {
  const filterBy = {};

  filters.forEach((filter) => {
    filterBy[`${filter.id}`] = filter.value;
    return filterBy;
  });

  return filterBy;
}

export function generateOrderByValues(sorted: Array<{ [string]: string }>): Array<string> {

  return sorted.map((sort) => {
    const name = snakeCase(sort.id).toUpperCase();
    const order = sort.desc ? 'ASC' : 'DESC';

    return `${name}_${order}`;
  });
}

export function paginationOptions({
  page, pageSize, filtered, sorted,
}: { page: number, pageSize: number, filtered: Array<{ [string]: string }>, sorted: Array<{ [string]: string }>}) {
  return {
    offset: page * pageSize,
    first: pageSize,
    condition: generateFilterValues(filtered),
    orderBy: sorted.length ? generateOrderByValues(sorted) : [],
  };
}

export function bytesToText(bytes: number = 0): string {
  let result = bytes;
  const gap = 1024;
  if (Math.abs(bytes) < gap) {
    return `${result} B`;
  }
  const units = ['KB', 'MB', 'GB', 'TB'];
  let unitIndex = -1;
  do {
    result /= gap;
    unitIndex += 1;
  } while (Math.abs(result) >= gap && unitIndex < units.length - 1);
  return `${result.toFixed(1)} ${units[unitIndex]}`;
}

export function identity(value: any) {
  return value;
}

// hashMapByUniqueField will return object of objects from an array of objects 'items'.
// Each object passed in will be indexed by key in this new object.
// Should only be used if 'field' specified is guaranteed to be unique among all objects in items
// (eg. the id field is usually unique), otherwise a second item will overwrite the first.
export function hashMapByUniqueField(items: Array<any>, field: string) {
  if (items) {
    const obj = {};
    items.forEach((item) => {
      obj[item[field]] = item;
    });
    return obj;
  }
  return {};
}

// stringTruncateWithEllipsis takes in a string, and if it has length greater
// than maxChars, will truncate it to that length minus 3 and add an ellipsis.
// Therefore, maxChars specifies the maximum number of characters to expect in the
// output, including the ellipsis, if added.
export function stringTruncateWithEllipsis(targetString: string, maxChars: number): string {
  if (!targetString) {
    return '';
  }
  return (targetString.length <= maxChars)
    ? targetString
    : `${targetString.substring(0, maxChars - 3)}...`;
}
