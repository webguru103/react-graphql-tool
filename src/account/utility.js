import { IDENTITY, ROLE_CATEGORY } from '../constants';

export function getRedirectPathByRoleCategory(roleCategory: string) {

  let redirectPath = '/';

  if (roleCategory === ROLE_CATEGORY.AGENT) {
    redirectPath = '/agent/sessions';
  } else if (roleCategory === ROLE_CATEGORY.SUPER_ADMIN || roleCategory === ROLE_CATEGORY.ADMIN) {
    redirectPath = '/admin/admin-management';
  } else if (roleCategory === ROLE_CATEGORY.CP_SEED_ADMIN
    || roleCategory === ROLE_CATEGORY.CP_SUPER_ADMIN
    || roleCategory === ROLE_CATEGORY.CP_ADMIN
  ) {
    redirectPath = 'cp-user/brokerage-management';
  }

  return redirectPath;
}

export function getRedirectPathBySelectedIdentity(redirectUrl: ?string, selectedIdentity: string, firstLogin: boolean) {
  if (redirectUrl) {
    return redirectUrl;
  }

  if (selectedIdentity === IDENTITY.CP_USER) {
    return '/cp-user';
  }

  if (selectedIdentity === IDENTITY.ADMIN) {
    return '/admin';
  }

  if (selectedIdentity === IDENTITY.AGENT) {
    if (firstLogin) {
      return '/agent/transactions?popup=true'; // TODO: Fix this, should be on account settings page.
    }
    return '/agent/sessions';
  }

  return '/';
}
