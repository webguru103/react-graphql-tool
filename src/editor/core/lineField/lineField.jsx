import * as React from 'react';
import { DragSource } from 'react-dnd';
import { getEmptyImage } from 'react-dnd-html5-backend';
import { withStyles } from '@material-ui/core';
import classnames from 'classnames';
import ResizeIcon from '@material-ui/icons/Brightness1';
import onClickOutside from 'react-onclickoutside';
import debounce from 'lodash.debounce';
import { compose, isDefined } from '../../../utility';
import styles from './lineField.styles';
import c, { EDITOR_MODE } from '../../constants';
import { adjustLineAttributesForPageSize } from '../utils';
import type { LineAttributes, AdaptedLineField } from '../flowTypes';

const MINIMUM_LINE_THICKNESS = 1;

/**
 * Specifies the drag source contract.
 * Only `beginDrag` function is required.
 */
const fieldSource = {
  beginDrag(props) {
    const {
      field: {
        id, x1, y1, x2, y2,
      }, inActiveSelection, pageIndex,
    } = props;
    return {
      id, pageIndex, x1, y1, x2, y2, inActiveSelection,
    };
  },
};

/**
 * Specifies resize dragging contract for vertex 1
 */
const fieldSourceResize1 = {
  beginDrag(props) {
    const {
      field: {
        id, x1, y1,
      }, pageIndex,
    } = props;
    return {
      id, pageIndex, x1, y1,
    };
  },
};

/**
 * Specifies resize dragging contract for vertex 2
 */
const fieldSourceResize2 = {
  beginDrag(props) {
    const {
      field: {
        id, x2, y2,
      }, pageIndex,
    } = props;
    return {
      id, pageIndex, x2, y2,
    };
  },
};

/**
 * Props to inject for dragging.
 */
function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    connectDragPreview: connect.dragPreview(),
    isDragging: monitor.isDragging(),
  };
}

/**
 * Props to inject for resize dragging on vertex 1.
 */
function collectResize1(connect, monitor) {
  return {
    connectDragPreviewResize1: connect.dragPreview(),
    connectDragSourceResize1: connect.dragSource(),
    isDraggingResize1: monitor.isDragging(),
  };
}

/**
 * Props to inject for resize dragging on vertex 2.
 */
function collectResize2(connect, monitor) {
  return {
    connectDragPreviewResize2: connect.dragPreview(),
    connectDragSourceResize2: connect.dragSource(),
    isDraggingResize2: monitor.isDragging(),
  };
}

type PropType = {
  classes: Object,
  connectDragSource: Function,
  connectDragSourceResize1: Function,
  connectDragSourceResize2: Function,
  connectDragPreview: Function,
  connectDragPreviewResize1: Function,
  connectDragPreviewResize2: Function,
  dragging: boolean,
  currentActivePageField: boolean,
  inActiveSelection: boolean,
  handleActiveSelectionRemove: Function,
  handleActiveSelectionClear: Function,
  offsetHeight: number,
  offsetWidth: number,
  addFieldsToSelection?: Function,
  field: AdaptedLineField,
  pageIndex: number,
  handleLineFieldUpdate: Function,
  pageRef: React.Element,
  isDragging: boolean,
  isDraggingResize1: boolean,
  isDraggingResize2: boolean,
  mode: string,
  zoom: number,
}

const UNBLOCK_DELAY = 250;
const UNBLOCK_TIMER_DEBOUNCE = 100;

export class LineFieldC extends React.Component<PropType, *> {

  // Store calculated values on 'this' rather than in state to
  // prevent double state updates on calculation in render() which
  // would occur with this.setState(), but still want to keep a record
  // of the last updated value on the object itself.
  queuedLineCoordinateUpdate = false;

  // Set to true to temporarily block updates from props
  // This should be done so that in the time between when the coordinates
  // for the line are updated at the top level and the time dragging stops,
  // there is no update back to the previous state, which may cause the line to flash.
  blockingUpdateAfterDrag = false;

  adjustedx1 = (this.props.field.x1 || 0) * this.props.zoom;

  adjustedy1 = (this.props.field.y1 || 0) * this.props.zoom;

  adjustedx2 = (this.props.field.x2 || 0) * this.props.zoom;

  adjustedy2 = (this.props.field.y2 || 0) * this.props.zoom;

  // Call this function to set block updates back to false
  // after a period of time.
  acceptUpdatesAfterTimer = debounce(() => {
    setTimeout(() => {
      this.blockingUpdateAfterDrag = false;
    }, UNBLOCK_DELAY);
  }, UNBLOCK_TIMER_DEBOUNCE)

  handleUpdateLine = debounce((adjusted: LineAttributes) => {
    const {
      field: { id },
      pageIndex,
      handleLineFieldUpdate,
    } = this.props;

    handleLineFieldUpdate(id, adjusted.x1, adjusted.y1, adjusted.x2, adjusted.y2, pageIndex);
  }, 200)

  componentDidMount() {
    this.props.connectDragPreview(getEmptyImage(), {
      captureDraggingState: false,
    });
    this.props.connectDragPreviewResize1(getEmptyImage(), {
      captureDraggingState: true,
    });
    this.props.connectDragPreviewResize2(getEmptyImage(), {
      captureDraggingState: true,
    });
  }

  shouldComponentUpdate(nextProps: PropType) {
    // Update this component when it is in the active selection,
    // unless just switching from selected to not selected.
    if (this.props.inActiveSelection !== nextProps.inActiveSelection) {
      return true;
    }

    // Update the component as a result of external events, like zoom change
    if (this.props.zoom !== nextProps.zoom) {
      return true;
    }

    if (!nextProps.inActiveSelection) {
      return false;
    }

    return true;
  }

  handleClickOutside = () => {
    this.props.handleActiveSelectionClear();
  };

  handleFieldClick = (ev: SyntheticMouseEvent<HTMLElement>, id: number) => {
    const {
      inActiveSelection,
      addFieldsToSelection,
      handleActiveSelectionRemove,
    } = this.props;

    ev.stopPropagation();

    if (ev.ctrlKey || ev.metaKey) {
      if (inActiveSelection) {
        handleActiveSelectionRemove(id);
      } else if (addFieldsToSelection) {
        // calling addFieldToSelection function with top/left corner
        // coordinates of a line rectangle
        addFieldsToSelection([id]);
      }
    } else if (!inActiveSelection && addFieldsToSelection) {
      addFieldsToSelection([id], { resetSelection: true });
    }
  };

  render() {
    const {
      field: {
        id, strokeThickness, temporary, x1, y1, x2, y2, positionLock,
      },
      classes,
      connectDragSource,
      dragging,
      currentActivePageField,
      inActiveSelection,
      pageRef,
      offsetHeight,
      offsetWidth,
      isDragging,
      isDraggingResize2,
      isDraggingResize1,
      connectDragSourceResize1,
      connectDragSourceResize2,
      mode,
      zoom,
    } = this.props;

    const pageRect = pageRef && pageRef.getBoundingClientRect();
    const strokeThicknessWithZoom = ((strokeThickness || 0) * zoom);

    // Adjust coordinates for each drag type.
    // Set queuedLineCoordinateUpdate = true to signal that an update should be sent to
    // handleUpdateLine and templateEditor function handleLineFieldUpdate
    if (!positionLock && isDraggingResize1 && isDefined(offsetWidth) && isDefined(offsetHeight)) {
      const updated = {
        x1: Math.round((x1 || 0) * zoom + offsetWidth),
        y1: Math.round((y1 || 0) * zoom + offsetHeight),
      };
      const adjusted = adjustLineAttributesForPageSize(pageRect.height, pageRect.width, updated);
      this.adjustedx1 = adjusted.x1;
      this.adjustedy1 = adjusted.y1;
      this.queuedLineCoordinateUpdate = true;
      this.blockingUpdateAfterDrag = true;
      this.acceptUpdatesAfterTimer();
    }

    if (!positionLock && isDraggingResize2 && isDefined(offsetWidth) && isDefined(offsetHeight)) {
      const updated = {
        x2: Math.round((x2 || 0) * zoom + offsetWidth),
        y2: Math.round((y2 || 0) * zoom + offsetHeight),
      };
      const adjusted = adjustLineAttributesForPageSize(pageRect.height, pageRect.width, updated);
      this.adjustedx2 = adjusted.x2;
      this.adjustedy2 = adjusted.y2;
      this.queuedLineCoordinateUpdate = true;
      this.blockingUpdateAfterDrag = true;
      this.acceptUpdatesAfterTimer();
    }

    // If coordinates have been updated and dragging stopped, trigger handleUpdateLine
    // to push changes up to parent components
    if (this.queuedLineCoordinateUpdate
      && !isDragging
      && !isDraggingResize2
      && !isDraggingResize2
      && !isDefined(offsetWidth)
      && !isDefined(offsetHeight)
    ) {
      this.queuedLineCoordinateUpdate = false;
      this.handleUpdateLine({
        x1: Math.floor(this.adjustedx1 / zoom),
        y1: Math.floor(this.adjustedy1 / zoom),
        x2: Math.floor(this.adjustedx2 / zoom),
        y2: Math.floor(this.adjustedy2 / zoom),
      });
      this.blockingUpdateAfterDrag = true;
      this.acceptUpdatesAfterTimer();
    }

    // If new props are received when not dragging, update directly
    if (!this.blockingUpdateAfterDrag
      && !isDragging
      && !isDraggingResize2
      && !isDraggingResize2
      && !isDefined(offsetWidth)
      && !isDefined(offsetHeight)
    ) {
      this.adjustedx1 = (this.props.field.x1 || 0) * zoom;
      this.adjustedy1 = (this.props.field.y1 || 0) * zoom;
      this.adjustedx2 = (this.props.field.x2 || 0) * zoom;
      this.adjustedy2 = (this.props.field.y2 || 0) * zoom;
    }

    // Trig Time
    // Line is rendered with a div, and a div is rendered with a top, left absolute positioning
    // and then rotated by degrees on its midpoint.
    // Therefore, need to translate x1, y1, x2, y2 coordinates into top, left and rotation degree to render the div correctly.
    const calculatedLineLength = Math.sqrt(((this.adjustedx1 - this.adjustedx2) ** 2) + ((this.adjustedy1 - this.adjustedy2) ** 2));
    const midPoint = {
      x: (this.adjustedx1 + this.adjustedx2) / 2,
      y: (this.adjustedy1 + this.adjustedy2) / 2,
    };
    const opposite = Math.abs(this.adjustedy1 - this.adjustedy2) / 2;
    const hypoteneuse = calculatedLineLength / 2;
    const oppositeOverHypoteneuse = opposite / hypoteneuse;
    const radians = Math.asin(oppositeOverHypoteneuse);
    let angle = radians * (180 / Math.PI);

    // isUpperPointTopRight is true if the vertex closest to the top of the page is further right
    // than the second vertex of the line.
    // The angle calculation above is found simply by using the absolute value of (y1 - y2) then divided by 2 to find
    // the length of the opposite side of a right triangle in which the hypoteneuse = calculatedLineLength / 2,
    // so this is basically assuming the line runs diagonally from the top left to the bottom right.
    // If isUpperPointTopRight is true, that means the line runs top right to bottom left,
    // and the angle calculation needs to be 'reversed' by finding the difference between the
    // calculated angle and 180.
    const uppermostPoint = this.adjustedy1 < this.adjustedy2 ? 'adjustedy1' : 'adjustedy2';
    const isUpperPointTopRight = (uppermostPoint === 'adjustedy1' && this.adjustedx1 > this.adjustedx2)
      || (uppermostPoint === 'adjustedy2' && this.adjustedx2 > this.adjustedx1);

    if (isUpperPointTopRight) {
      angle = Math.abs(angle - 180);
    }

    const topLeftCorner = {
      x: midPoint.x - (calculatedLineLength / 2),
      y: midPoint.y,
    };

    return (
      <div className="fieldOutClick">
        { (mode !== EDITOR_MODE.TEMPLATE_PUBLISHED && positionLock !== true) ? connectDragSource && connectDragSource((
          <div
            onMouseDown={e => this.handleFieldClick(e, id)}
            role="button"
            tabIndex="-1"
            data-testid="line-field-container"
            style={{
              height: 0,
              width: calculatedLineLength,
              transform: `translate(${topLeftCorner.x}px, ${topLeftCorner.y}px) rotate(${angle}deg)`,
            }}
            key={id}
            className={classnames({
              [classes.fieldContainer]: true,
              [classes.dragging]: dragging && inActiveSelection,
              [classes.temporary]: temporary,
            })}
          >
            <div
              style={{
                height: strokeThicknessWithZoom < MINIMUM_LINE_THICKNESS ? MINIMUM_LINE_THICKNESS : strokeThicknessWithZoom,
                width: calculatedLineLength,
              }}
              className={classnames({
                [classes.field]: true,
                [classes.activeSelection]: inActiveSelection,
              })}
            />
          </div>
        ))
          : (
            <div
              onMouseDown={e => this.handleFieldClick(e, id)}
              role="button"
              tabIndex="-1"
              data-testid="line-field-container"
              style={{
                height: 0,
                width: calculatedLineLength,
                transform: `translate(${topLeftCorner.x}px, ${topLeftCorner.y}px) rotate(${angle}deg)`,
              }}
              key={id}
              className={classnames({
                [classes.fieldContainer]: true,
                [classes.temporary]: temporary,
              })}
            >
              <div
                style={{
                  height: strokeThicknessWithZoom < MINIMUM_LINE_THICKNESS ? MINIMUM_LINE_THICKNESS : strokeThicknessWithZoom,
                  width: calculatedLineLength,
                }}
                className={classnames({
                  [classes.field]: true,
                  [classes.activeSelection]: inActiveSelection,
                })}
              />
            </div>
          )
        }

        {(mode !== EDITOR_MODE.TEMPLATE_PUBLISHED && positionLock !== true) && currentActivePageField && connectDragSourceResize1((
          <div
            onMouseDown={e => e.stopPropagation()}
            onKeyDown={e => e.stopPropagation()}
            role="button"
            tabIndex="-1"
          >
            <ResizeIcon
              className={classnames({
                [classes.resizeHandle]: true,
                [classes.dragging]: dragging,
              })}
              style={{
                left: this.adjustedx1,
                top: this.adjustedy1,
              }}
            />
          </div>
        ))
        }
        {(mode !== EDITOR_MODE.TEMPLATE_PUBLISHED && positionLock !== true) && currentActivePageField && connectDragSourceResize2((
          <div
            onMouseDown={e => e.stopPropagation()}
            onKeyDown={e => e.stopPropagation()}
            role="button"
            tabIndex="-1"
          >
            <ResizeIcon
              className={classnames({
                [classes.resizeHandle]: true,
                [classes.dragging]: dragging,
              })}
              style={{
                left: this.adjustedx2,
                top: this.adjustedy2,
              }}
            />
          </div>
        ))
        }
      </div>
    );
  }

}

export default compose(
  DragSource(c.ItemTypes.FIELD, fieldSource, collect),
  DragSource(c.ItemTypes.RESIZE_LINE, fieldSourceResize1, collectResize1),
  DragSource(c.ItemTypes.RESIZE_LINE, fieldSourceResize2, collectResize2),
  withStyles(styles, { withTheme: true }),
  onClickOutside,
)(LineFieldC);
