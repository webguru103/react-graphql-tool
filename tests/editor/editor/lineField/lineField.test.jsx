/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import 'jest-dom/extend-expect';
import { render, cleanup } from 'react-testing-library';
import { withContext } from '../../../test-helpers/contextWrapper';
import { LineFieldC } from '../../../../src/editor/core/lineField/lineField';
import { noop } from '../../../../src/utility';

describe('LineField', () => {
  afterEach(cleanup);
  const Component = withContext(LineFieldC);

  test('renders a div with 90 degree rotation', () => {

    const field = {
      id: 53151,
      fieldName: 'LineField',
      x1: 10,
      y1: 10,
      x2: 10,
      y2: 30,
      strokeThickness: 5,
    };

    const pageRects = {
      0: {
        width: 900,
        height: 1200,
      },
    };

    const { container, getByTestId } = render(<Component
      classes={{}}
      connectDragSource={x => x}
      connectDragPreview={x => x}
      isDragging={false}
      connectDragPreviewResize1={x => x}
      connectDragSourceResize1={x => x}
      isDraggingResize1={false}
      connectDragPreviewResize2={x => x}
      connectDragSourceResize2={x => x}
      isDraggingResize2={false}
      dragging={false}
      resizing={false}
      offsetHeight={null}
      offsetWidth={null}
      field={field}
      key={field.id}
      pageRects={pageRects}
      pageNumber={0}
      handleActivePageField={noop}
      handleLineFieldUpdate={noop}
      zoom={1}
      handleActiveSelectionClear={noop}
      addFieldToSelection={noop}
      handleActiveSelectionRemove={noop}
      currentActivePageField={false}
      inActiveSelection={false}
      outsideClickIgnoreClass="fieldOutClick"
    />);

    expect(container).toMatchSnapshot();
    const lineField = getByTestId('line-field-container');
    expect(lineField.style._values.transform).toBe('translate(0px, 20px) rotate(90deg)');
  });

  test('renders a div with 51.11... degree rotation', () => {

    const field = {
      id: 52152,
      fieldName: 'LineField',
      x1: 42,
      y1: 52,
      x2: 263,
      y2: 326,
      strokeThickness: 5,
    };

    const pageRects = {
      0: {
        width: 900,
        height: 1200,
      },
    };

    const { getByTestId } = render(<Component
      classes={{}}
      connectDragSource={x => x}
      connectDragPreview={x => x}
      isDragging={false}
      connectDragPreviewResize1={x => x}
      connectDragSourceResize1={x => x}
      isDraggingResize1={false}
      connectDragPreviewResize2={x => x}
      connectDragSourceResize2={x => x}
      isDraggingResize2={false}
      dragging={false}
      resizing={false}
      offsetHeight={null}
      offsetWidth={null}
      field={field}
      key={field.id}
      pageRects={pageRects}
      pageNumber={0}
      handleActivePageField={noop}
      handleLineFieldUpdate={noop}
      zoom={1}
      handleActiveSelectionClear={noop}
      addFieldToSelection={noop}
      handleActiveSelectionRemove={noop}
      currentActivePageField={false}
      inActiveSelection={false}
      outsideClickIgnoreClass="fieldOutClick"
    />);

    const lineField = getByTestId('line-field-container');
    expect(lineField.style._values.transform).toBe('translate(-23.509232712377894px, 189px) rotate(51.11141655726794deg)');
  });

  test('renders a div with 45 degree rotation', () => {

    const field = {
      id: 53151,
      fieldName: 'LineField',
      x1: 10,
      y1: 10,
      x2: 30,
      y2: 30,
      strokeThickness: 5,
    };

    const pageRects = {
      0: {
        width: 900,
        height: 1200,
      },
    };

    const { getByTestId } = render(<Component
      classes={{}}
      connectDragSource={x => x}
      connectDragPreview={x => x}
      isDragging={false}
      connectDragPreviewResize1={x => x}
      connectDragSourceResize1={x => x}
      isDraggingResize1={false}
      connectDragPreviewResize2={x => x}
      connectDragSourceResize2={x => x}
      isDraggingResize2={false}
      dragging={false}
      resizing={false}
      offsetHeight={null}
      offsetWidth={null}
      field={field}
      key={field.id}
      pageRects={pageRects}
      pageNumber={0}
      handleActivePageField={noop}
      handleLineFieldUpdate={noop}
      zoom={1}
      handleActiveSelectionClear={noop}
      addFieldToSelection={noop}
      handleActiveSelectionRemove={noop}
      currentActivePageField={false}
      inActiveSelection={false}
      outsideClickIgnoreClass="fieldOutClick"
    />);

    const lineField = getByTestId('line-field-container');
    expect(lineField.style._values.transform).toBe('translate(5.857864376269049px, 20px) rotate(44.99999999999999deg)');
  });
});
