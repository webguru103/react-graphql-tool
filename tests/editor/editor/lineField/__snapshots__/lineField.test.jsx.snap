// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineField renders a div with 90 degree rotation 1`] = `
<div>
  <div
    class="fieldOutClick"
  >
    <div
      class=""
      data-testid="line-field-container"
      role="button"
      style="height: 0px; width: 20px; transform: translate(0px, 20px) rotate(90deg);"
      tabindex="-1"
    >
      <div
        class=""
        style="height: 5px; width: 20px;"
      />
    </div>
  </div>
</div>
`;
